test/Main.hs

```diff
@@ -316,6 +316,26 @@ autocommitTest td = testCase "Autocommit mode works" .
     assertEqualEq "Other connection sees autocommited data" 1 n
   runQuery_ $ rawSQL "DELETE FROM test1_ WHERE a = $1" sint
 
+preparedStatementTest :: TestData -> Test
+preparedStatementTest td = testCase "Execution of prepared statements works" .
+                           runTestEnv td defaultTransactionSettings $ do
+  let i1 = 42 :: Int32
+  runPreparedQuery_ "select1" $ "SELECT" <?> i1
+  o1 <- fetchOne runIdentity
+  assertEqualEq "Results match" i1 o1
+
+  let i2 = 89 :: Int32
+  runPreparedQuery_ "select1" $ "SELECT" <?> i2
+  o2 <- fetchOne runIdentity
+  assertEqualEq "Results match" i2 o2
+
+  let i3 = "lalala" :: String
+  -- Changing parameter type in an already prepared statement shouldn't work.
+  o3 <- try $ runPreparedQuery_ "select1" $ "SELECT" <?> i3
+  case o3 of
+    Left DBException{} -> pure ()
+    Right r3           -> liftBase . assertFailure $ "Expected DBException, but got" <+> show r3
+
```

Comment:
`runPreparedQuery01` is not covered by the tests.

Reply:
Right, but so isn't runQuery01 ;) I know, a somewhat poor excuse.

Reply:
Yeah, very poor :grin: